mergebot_backend/merge.py

```diff
@@ -289,6 +319,17 @@ def merge_git_pr(self, pr):
             _set_up(tmp_dir)
             self.execute_merge_lifecycle(pr, tmp_dir, pr_logger)
         except AssertionError as exc:
+            pr.post_commit_status(
+                state=github_helper.COMMIT_STATE_FAILURE,
```

Comment:
AssertionErrors indicate something has gone preventably wrong in the PR (as mentioned in the above comment) and thus FAILURE is the appropriate level.